### PR TITLE
at86rf2xx: codesize optimizations

### PR DESCRIPTION
**Pull request by OlegHahm**
I picked up and adopted #4419 since @thomaseichinger apparently quit working on this. Needs careful review and thorough testing since there was quite some rebasing required.

### CODE REVIEW

File: drivers/at86rf2xx/at86rf2xx_netdev.c [hunk 1]
```diff
@@ -398,95 +376,47 @@ static int _set(netdev2_t *netdev, netopt_t opt, void *val, size_t len)
...
-        case NETOPT_NID:
-            if (len > sizeof(uint16_t)) {
-                res = -EOVERFLOW;
-            }
-            else {
-                at86rf2xx_set_pan(dev, *((uint16_t *)val));
-                /* don't set res to set netdev2_ieee802154_t::pan */
-            }
-            break;
-
```
Comment (PeterKietzmann) [inline drivers/at86rf2xx/at86rf2xx_netdev.c]: Where are the addresses/ the pan actually written to the device?

File: drivers/at86rf2xx/at86rf2xx_netdev.c [hunk 2]
```diff
@@ -398,95 +376,47 @@ static int _set(netdev2_t *netdev, netopt_t opt, void *val, size_t len)
...
-                    break;
-                }
-                at86rf2xx_set_chan(dev, chan);
-                /* don't set res to set netdev2_ieee802154_t::chan */
-            }
+            assert(len != sizeof(uint16_t));
+            uint8_t chan = ((uint8_t *)val)[0];
+            if (chan < AT86RF2XX_MIN_CHANNEL ||
+                chan > AT86RF2XX_MAX_CHANNEL) {
+                res = -ENOTSUP;
```
Comment (PeterKietzmann) [inline drivers/at86rf2xx/at86rf2xx_netdev.c]: Why not staying with `EINVAL`?

File: drivers/at86rf2xx/at86rf2xx_netdev.c [hunk 3]
```diff
@@ -548,9 +474,11 @@ static int _set(netdev2_t *netdev, netopt_t opt, void *val, size_t len)
             break;
 
         case NETOPT_CSMA_RETRIES:
-            if ((len > sizeof(uint8_t)) ||
-                (*((uint8_t *)val) > 5)) {
-                res = -EOVERFLOW;
+            assert(len <= sizeof(uint8_t));
+            if( !(dev->netdev.flags & AT86RF2XX_OPT_CSMA ||
+                (*((uint8_t *)val) > 5)) ) {
```
Comment (PeterKietzmann) [inline drivers/at86rf2xx/at86rf2xx_netdev.c]: Maybe we should spend a definition for the max. amount of retires?

File: drivers/at86rf2xx/at86rf2xx_netdev.c (hunk 1 above)
Comment (OlegHahm) [inline drivers/at86rf2xx/at86rf2xx_netdev.c]: Phew, I think something went wrong during the rebase.

**Comment by thomaseichinger** [comment]
@OlegHahm updated #4419. As mentioned, I didn't find time to test it yet though.

**Comment by OlegHahm** [comment]
I think something went wrong rebasing here.

**Pull request closed without merging**